packages/strapi-plugin-graphql/hooks/graphql/index.js

```diff
@@ -95,6 +95,21 @@ module.exports = strapi => {
         schemaDef.resolvers = resolvers;
       }
 
+      const serverConfigOmittedOptions = [
+        'shadowCRUD',
+        'playgroundAlways',
+        'depthLimit',
+        'amountLimit',
+      ];
+      const serverConfig = _.omit(
+        _.get(strapi.plugins.graphql, 'config', {}),
+        serverConfigOmittedOptions
+      );
+
+      serverConfig.engine = _.get(serverConfig, 'engine', false);
```

Comment:
you can put those as defaults in the serverParams to make it even simpler

Reply:
Done!